enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/EndpointSecurityUtils.java

```diff
@@ -34,30 +33,29 @@
      * Adds the backend endpoint security header to the given requestContext.
      *
      * @param requestContext requestContext instance to add the backend endpoint security header
-     * @param apiKeyValidationInfoDTO apiKeyValidationInfoDTO containing necessary info
      */
-    public static void addEndpointSecurity(RequestContext requestContext,
-                                                    APIKeyValidationInfoDTO apiKeyValidationInfoDTO) {
+    public static void addEndpointSecurity(RequestContext requestContext) {
         SecurityInfo securityInfo;
-        if (apiKeyValidationInfoDTO.getType() != null &&
-                requestContext.getMatchedAPI().getEndpointSecurity() != null) {
-            if (apiKeyValidationInfoDTO.getType().equals(APIConstants.API_KEY_TYPE_PRODUCTION)) {
-                securityInfo = requestContext.getMatchedAPI().getEndpointSecurity().
-                        getProductionSecurityInfo();
-            } else {
-                securityInfo = requestContext.getMatchedAPI().getEndpointSecurity().
-                        getSandBoxSecurityInfo();
-            }
-            if (securityInfo != null && securityInfo.isEnabled() &&
-                    APIConstants.AUTHORIZATION_HEADER_BASIC.
-                            equalsIgnoreCase(securityInfo.getSecurityType())) {
-                requestContext.getRemoveHeaders().remove(APIConstants.AUTHORIZATION_HEADER_DEFAULT
-                        .toLowerCase());
-                requestContext.addOrModifyHeaders(APIConstants.AUTHORIZATION_HEADER_DEFAULT,
-                        APIConstants.AUTHORIZATION_HEADER_BASIC + ' ' +
-                                Base64.getEncoder().encodeToString((securityInfo.getUsername() +
-                                        ':' + String.valueOf(securityInfo.getPassword())).getBytes()));
-            }
+        String keyType = "";
+        if (requestContext.getAuthenticationContext() != null) {
+            keyType = requestContext.getAuthenticationContext().getKeyType();
+        }
+        if (APIConstants.API_KEY_TYPE_PRODUCTION.equals(keyType)) {
+            securityInfo = requestContext.getMatchedAPI().getEndpointSecurity().
+                    getProductionSecurityInfo();
```

Comment:
Shall we add the null checks for endpoint security?

Reply:
before getting by the type. like in https://github.com/wso2/product-microgateway/pull/2455/files#diff-bd730bb927f90647f7c2a16891c9af03bd4e58bd36b12493a187f52c1046e66dL43